From 509aec4fc5464431e7b49fd0c44ded6f1f01ee37 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 29 Apr 2021 19:50:26 +0300
Subject: [PATCH] [SW-0] fwupd: call rma_erase() directly

No need for additional function that only returns the real one.
Call rma_erase() directly.

Change-Id: Ief721e4a9be575c7d47b38cb8ec51a32334ed332
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 8228cf5..dd3fff1 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -179,11 +179,6 @@
 	return 1;
 }
 
-static int module_fwupd_load_type_rma_erase(void)
-{
-	return rma_erase();
-}
-
 static int module_fwupd_load_type_run(void)
 {
 	state.is_fit_checked = !fit_load_images();
@@ -206,7 +201,7 @@
 	case FIT_LOAD_TYPE_NIC_EN_MASK:
 		return module_fwupd_load_type_nic_en(fit);
 	case FIT_LOAD_TYPE_RMA_ERASE:
-		return module_fwupd_load_type_rma_erase();
+		return rma_erase();
 	case FIT_LOAD_TYPE_RUN:
 		return module_fwupd_load_type_run();
 	default:
